scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -638,6 +674,21 @@ def get_containers( self, running_or_created_after=None, glob_list=None, k8s_cac
                 k8s_info['pod_namespace'] = pod_namespace
                 k8s_info['k8s_container_name'] = cname
 
+                # build the path to the log file on disk
+                log_path = "%s/%s_%s_%s-%s.log" % (self._log_base, pod_name, pod_namespace, cname, cid)
+
+                if self._query_filesystem:
+                    # get pod uid from the path
+                    # the log path should be a link to the /var/logs/pod/<pod_uid>/N.log
+                    # so read the link.  Note, use os.readlink rather than os.path.realpath
+                    # because we only want to traverse one level of links, rather than finding
+                    # the eventual path on disk (which may not be the /var/logs/pod/<pod_uid>/N.log)
+                    if os.path.islink( log_path ):
+                        link = os.readlink( log_path )
+                        m = self._pod_re.match( link )
+                        if m and cname == m.group(2):
+                            k8s_info['pod_uid'] = m.group(1)
+
```

Comment:
Does both the kublet api and filesystem approach not return the pod uid?

Reply:
The filesystem approach doesn't - this is the filesystem approach for getting the pod uid.

However you are right an it looks like the kubelet api is not currently returning the pod uid.  I can move this in to the `_get_containers_from_filesystem` and add pod uid information in to the call to `_get_containers_from_kubelet`.

Reply:
Ok, we can do that later.

Reply:
Filed https://scalyr.myjetbrains.com/youtrack/issue/AGENT-151